fsi/fsi.go

```diff
@@ -253,6 +275,70 @@ func removeLinkFile(dir string) error {
 	return os.Remove(dir)
 }
 
+func isLowValueFile(f os.FileInfo) bool {
+	if f != nil {
+		filesToBeRemoved := []string{
+			// generic files
+			"^\\..*\\.swp$", // Swap file for vim state
+
+			// macOS specific files
+			"^\\.DS_Store$", // Stores custom folder attributes
+			"^\\.AppleDouble$", // Stores additional file resources
+			"^\\.LSOverride$", // Contains the absolute path to the app to be used
+			"^Icon\\r$", // Custom Finder icon: http://superuser.com/questions/298785/icon-file-on-os-x-desktop
+			"^\\._.*", // Thumbnail
+			"^\\.Spotlight-V100(?:$|\\/)", // Directory that might appear on external disk
+			"\\.Trashes", // File that might appear on external disk
+			"^__MACOSX$", // Resource fork
+
+			// linux specific files
+			"~$", // Backup file
+
+			// Windows specific files
+			"^Thumbs\\.db$", // Image file cache
+			"^ehthumbs\\.db$", // Folder config file
+			"^Desktop\\.ini$", // Stores custom folder attributes
+			"@eaDir$", // Synology Diskstation "hidden" folder where the server stores thumbnails
+		}
+
+		pattern := strings.Join(filesToBeRemoved, "|")
+		matched, err := regexp.MatchString(pattern, filepath.Base(f.Name()))
+		if err != nil {
+			return false
+		}
+		if matched {
+			return true
+		}
+	}
+	return false
+}
+
+func getLowValueFiles(files *[]string) filepath.WalkFunc {
+	return func(path string, f os.FileInfo, err error) error {
```

Comment:
Make sure to run `go fmt ./...` in order to apply the standard golang code style guide.